api/src/pcapi/core/users/api.py

```diff
@@ -985,10 +985,13 @@ def get_eligibility_at_date(
         return None
 
     age = users_utils.get_age_at_date(date_of_birth, specified_datetime)
+    if age is None:
+        return None
 
     if age in constants.ELIGIBILITY_UNDERAGE_RANGE:
         return EligibilityType.UNDERAGE
-    if age == constants.ELIGIBILITY_AGE_18:
+    # If the user is older than 18 in UTC timezone, we consider them eligible until they reach eligibility_end
+    if constants.ELIGIBILITY_AGE_18 <= age < eligibility_end:
```

Comment:
if constants.ELIGIBILITY_AGE_18 <= age and specified_datetime < eligibility_end
```suggestion
    if constants.ELIGIBILITY_AGE_18 <= age and specified_datetime < eligibility_end:
```